### PR TITLE
minor improvements for FragmentFD downloader

### PR DESCRIPTION
**Pull request by flashdagger**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
* detect already downloaded file
* suppress console title output on HttpQuietDownloader
*  allow keyboard interrupt (Ctrl-C) when downloading fragments concurrently

### CODE REVIEW

**Comment by flashdagger** [comment]
**Note**: `HttpQuietDownloader.to_console_title()` will have no effect because of recent changes in `FileDownloader._set_ydl()`.

**Comment by pukkandan** [comment]
ah, right. I'll fix that

File: yt_dlp/downloader/fragment.py [hunk 1]
```diff
@@ -258,8 +264,8 @@ def frag_progress_hook(s):
             if ctx_id is not None and s.get('ctx_id') != ctx_id:
                 return
 
-            state['max_progress'] = ctx.get('max_progress')
-            state['progress_idx'] = ctx.get('progress_idx')
+                state['max_progress'] = ctx.get('max_progress')
```
Comment (flashdagger) [inline yt_dlp/downloader/fragment.py:267]: Indent is wrong here and must be reverted.

File: yt_dlp/downloader/fragment.py [hunk 2]
```diff
@@ -258,8 +264,8 @@ def frag_progress_hook(s):
...
+                state['progress_idx'] = ctx.get('progress_idx')
```
Comment (flashdagger) [inline yt_dlp/downloader/fragment.py:268]: Indent is wrong here and must be reverted.

File: yt_dlp/downloader/fragment.py [hunk 3]
```diff
@@ -290,7 +296,7 @@ def frag_progress_hook(s):
                 ctx['speed'] = state['speed'] = self.calc_speed(
                     ctx['fragment_started'], time_now, frag_downloaded_bytes)
                 ctx['prev_frag_downloaded_bytes'] = frag_downloaded_bytes
-            self._hook_progress(state, info_dict)
+                self._hook_progress(state, info_dict)
```
Comment (flashdagger) [inline yt_dlp/downloader/fragment.py:299]: Indent is wrong here and must be reverted.

**Comment by pukkandan** [comment]
I don't understand 4f40c26f12201fd0d7e0ffe6082a61466d022cc5

Could you explain how to reproduce the issue that it is supposed to solve?

**Comment by flashdagger** [comment]
> I don't understand [4f40c26](https://github.com/yt-dlp/yt-dlp/commit/4f40c26f12201fd0d7e0ffe6082a61466d022cc5)
> 
> Could you explain how to reproduce the issue that it is supposed to solve?

Sure. If you download a fragmented file again (after successful download) all the fragments are downloaded again and appended. So simply download a fragmented video twice with `--no-force-overwrites`.

**Comment by pukkandan** [comment]
Without that patch

```
❯ yt-dlp test:youtube_7 --no-force-over -f 139
[TestURL] Test URL: https://youtube.com/watch?v=Tq92D6wQ1mg
[youtube] Tq92D6wQ1mg: Downloading webpage
[youtube] Tq92D6wQ1mg: Downloading android player API JSON
[youtube] Tq92D6wQ1mg: Downloading tv embedded player API JSON
[youtube] Tq92D6wQ1mg: Downloading web embedded client config
[youtube] Tq92D6wQ1mg: Downloading player 9cdfefcf
[youtube] Tq92D6wQ1mg: Downloading web embedded player API JSON
[youtube] Tq92D6wQ1mg: Downloading MPD manifest
[info] Tq92D6wQ1mg: Downloading 1 format(s): 139
[dashsegments] Total fragments: 12
[download] Destination: [MMD] Adios - EVERGLOW [+Motion DL] [Tq92D6wQ1mg].m4a
[download] 100% of 632.98KiB in 00:00
[FixupM4a] Correcting container of "[MMD] Adios - EVERGLOW [+Motion DL] [Tq92D6wQ1mg].m4a"


❯ yt-dlp test:youtube_7 --no-force-over -f 139
[TestURL] Test URL: https://youtube.com/watch?v=Tq92D6wQ1mg
[youtube] Tq92D6wQ1mg: Downloading webpage
[youtube] Tq92D6wQ1mg: Downloading android player API JSON
[youtube] Tq92D6wQ1mg: Downloading tv embedded player API JSON
[youtube] Tq92D6wQ1mg: Downloading web embedded client config
[youtube] Tq92D6wQ1mg: Downloading player 9cdfefcf
[youtube] Tq92D6wQ1mg: Downloading web embedded player API JSON
[youtube] Tq92D6wQ1mg: Downloading MPD manifest
[info] Tq92D6wQ1mg: Downloading 1 format(s): 139
[download] [MMD] Adios - EVERGLOW [+Motion DL] [Tq92D6wQ1mg].m4a has already been downloaded
[download] 100% of 632.08KiB
```

**Comment by flashdagger** [comment]
I see. I also used `--no-part` which provokes this behavior:

```
$ yt-dlp test:youtube_7 --no-force-over --no-config --no-part -f 139
[TestURL] Test URL: https://youtube.com/watch?v=Tq92D6wQ1mg
[youtube] Tq92D6wQ1mg: Downloading webpage
[youtube] Tq92D6wQ1mg: Downloading android player API JSON
[youtube] Tq92D6wQ1mg: Downloading tv embedded player API JSON
[youtube] Tq92D6wQ1mg: Downloading web embedded client config
[youtube] Tq92D6wQ1mg: Downloading player 9cdfefcf
[youtube] Tq92D6wQ1mg: Downloading web embedded player API JSON
[youtube] Tq92D6wQ1mg: Downloading MPD manifest
[info] Tq92D6wQ1mg: Downloading 1 format(s): 139
[dashsegments] Total fragments: 12
[download] Destination: [MMD] Adios - EVERGLOW [+Motion DL] [Tq92D6wQ1mg].m4a
[download] 100% of 1.24MiB in 00:02
[FixupM4a] Correcting container of "[MMD] Adios - EVERGLOW [+Motion DL] [Tq92D6wQ1mg].m4a"
ERROR: Postprocessing: file:[MMD] Adios - EVERGLOW [+Motion DL] [Tq92D6wQ1mg].m4a: Invalid data found when processing input
```

**Comment by pukkandan** [comment]
https://github.com/yt-dlp/yt-dlp/commit/4f40c26f12201fd0d7e0ffe6082a61466d022cc5 is a bit complicated. We need to account for the various combinations of `--continue`, `--overwrite` and `--part`

https://github.com/yt-dlp/yt-dlp/pull/3498/commits/f4f9ff049e9dcaad80e466b3f39e6d5eec726018 has already been fixed

I'll cherry-pick just https://github.com/yt-dlp/yt-dlp/pull/3498/commits/f2046d9f030fa5d3a016c357bffe41cf0ffdb11f for now

**Comment by flashdagger** [comment]
> [f4f9ff0](https://github.com/yt-dlp/yt-dlp/commit/f4f9ff049e9dcaad80e466b3f39e6d5eec726018) has already been fixed
> 
Nearly :-D

**Comment by flashdagger** [comment]
@pukkandan Is the remaining change still relevant? Otherwise we can close this PR.

**Comment by pukkandan** [comment]
The issue https://github.com/yt-dlp/yt-dlp/pull/3498#issuecomment-1114132257 still exists, but these changes cannot be merged as-is

> [4f40c26](https://github.com/yt-dlp/yt-dlp/commit/4f40c26f12201fd0d7e0ffe6082a61466d022cc5) is a bit complicated. We need to account for the various combinations of `--continue`, `--overwrite` and `--part`

I left the PR open as basically a bug report. But feel free to close it if you don't really care about the issue. Nobody else has ever reported this, so I agree it is a "very special case"

**Comment by flashdagger** [comment]
closing as this is a negligible. nevertheless it was already partially merged. Thanks @pukkandan.

**Pull request closed without merging**